### PR TITLE
fix: remove checks for state for sentry mode

### PR DESCRIPTION
**Pull request by hobbe**
Similar to PR #64, and as requested in PR #61, remove check for current state in case it is not accurate.

### CODE REVIEW

**Comment by alandtse** [comment]
Given the [request](https://github.com/home-assistant/core/pull/32938) in the main HA, I'll leave this open for any further fixes we can batch together.

**Comment by hobbe** [comment]
Regarding the issue in HA PR https://github.com/home-assistant/core/pull/32938, we only know that sentry mode is available when we query `vehicle_state`, that is when `update()` is called.
This means that update needs to be called before any setup in HA.

**Comment by alandtse** [comment]
I think you can just set the [`entity_registry_enabled_default`](https://developers.home-assistant.io/docs/entity_index#advanced-properties) parameter to `False` on init and then enable it once the car is queried and we know.  Probably makes sense to add that parameter to vehicle.py as `True` for default. In sentry_mode.py you can then add the KeyError check on async_update.

**Comment by hobbe** [comment]
@alandtse , I have added a flag to know if sentry mode is available. If not, nothing should happen and sentry mode is always False.

As long as there is no flag at the vehicle level, then the switch will exist but do nothing. Can a switch be set to disabled?

File: teslajsonpy/sentry_mode.py [hunk 1]
```diff
@@ -43,11 +44,21 @@ def __init__(self, data, controller):
         last_update = self._controller.get_last_update_time(self._id)
         if last_update >= self.__manual_update_time:
             data = self._controller.get_state_params(self._id)
-            self.__sentry_mode = data and data["sentry_mode"]
+            self.__sentry_mode_available = data and data["sentry_mode_available"]
+            if self.__sentry_mode_available:
+                self.__sentry_mode = data["sentry_mode"]
+
+    def is_available(self):
```
Comment (alandtse) [inline teslajsonpy/sentry_mode.py]: Change this to `available` so [HA can use it directly](https://developers.home-assistant.io/docs/entity_index#generic-properties).

File: teslajsonpy/sentry_mode.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def __init__(self, data, controller):
         """
         super().__init__(data, controller)
         self.__manual_update_time = 0
+        self.__sentry_mode_available = False
```
Comment (alandtse) [inline teslajsonpy/sentry_mode.py]: Isn't this unnecessary since it's in vehicle?

File: teslajsonpy/sentry_mode.py [hunk 3]
```diff
@@ -43,11 +44,21 @@ def __init__(self, data, controller):
...
+            self.__sentry_mode_available = data and data["sentry_mode_available"]
+            if self.__sentry_mode_available:
+                self.__sentry_mode = data["sentry_mode"]
+
+    def is_available(self):
+        """Return whether the sentry mode is available."""
+        return self.__sentry_mode_available
+
+    def is_enabled(self):
+        """Return whether the sentry mode is enabled, always False if sentry mode is not available."""
```
Comment (alandtse) [inline teslajsonpy/sentry_mode.py]: HA wants you to return `None` typically for values, but I'm not sure for switches. Can you test HA with `None` to see what the behavior is for a switch?

Comment (hobbe) [reply teslajsonpy/sentry_mode.py]: OK, I will test this.

Comment (hobbe) [reply teslajsonpy/sentry_mode.py]: A None value behaves as a False. Change to `available` and `is_on as per HA`entity.

File: teslajsonpy/sentry_mode.py (hunk 1 above)
Comment (hobbe) [reply teslajsonpy/sentry_mode.py]: Changed to `available`.

**Comment by hobbe** [comment]
Argh... I am too late. I removed the unecessary `__sentry_mode_available`, as it is already available in VehicleDevice as `sentry_mode_available`. I keep it for next PR.

File: teslajsonpy/sentry_mode.py (hunk 2 above)
Comment (hobbe) [reply teslajsonpy/sentry_mode.py]: Correct, it is in VehicleDevice as `sentry_mode_available`. `__sentry_mode_available` can be removed.

**Comment by alandtse** [comment]
Ooops. I thought you were done since you addressed everything. If you know anything about using pytest, it'd be nice to have a suite of tests for sentry mode.  That may be a worthwhile PR to add and then we can remove it in that one too.

**Pull request merged**